sensorprocess/movementsensorprocess.go

```diff
@@ -13,85 +13,131 @@ import (
+						config.Logger.Warnw("Retrying odometer sensor reading due to error from cartofacade", "error", err)
+					}
+				} else {
+					odometerDone = true
 				}
-			} else {
+			}
+			if !imuDone {
+				if err := config.tryAddIMUReading(ctx, *reading.TimedIMUResponse); err != nil {
+					if !errors.Is(err, cartofacade.ErrUnableToAcquireLock) {
+						config.Logger.Warnw("Retrying IMU sensor reading due to error from cartofacade", "error", err)
+					}
+				} else {
+					imuDone = true
+				}
+			}
+			if imuDone && odometerDone {
 				return nil
 			}
 		}
 	}
 }
 
-// tryAddIMUReadingOnce adds a reading to the carto facade and does not retry. Returns remainder of time interval.
-func (config *Config) tryAddIMUReadingOnce(ctx context.Context, reading s.TimedIMUReadingResponse) int {
+// tryAddMovementSensorReadingOnce adds a reading to the carto facade and does not retry. Returns remainder of time interval.
+func (config *Config) tryAddMovementSensorReadingOnce(ctx context.Context, reading s.TimedMovementSensorReadingResponse) int {
 	startTime := time.Now().UTC()
-	if err := config.tryAddIMUReading(ctx, reading); err != nil {
-		if errors.Is(err, cartofacade.ErrUnableToAcquireLock) {
-			config.Logger.Debugw("Skipping sensor reading due to lock contention in cartofacade", "error", err)
-		} else {
-			config.Logger.Warnw("Skipping sensor reading due to error from cartofacade", "error", err)
+
+	if config.MovementSensor.Properties().OdometerSupported {
+		if err := config.tryAddOdometerReading(ctx, *reading.TimedOdometerResponse); err != nil {
+			if errors.Is(err, cartofacade.ErrUnableToAcquireLock) {
+				config.Logger.Debugw("Skipping odometer sensor reading due to lock contention in cartofacade", "error", err)
+			} else {
+				config.Logger.Warnw("Skipping odometer sensor reading due to error from cartofacade", "error", err)
+			}
 		}
 	}
+
+	if config.MovementSensor.Properties().IMUSupported {
+		if err := config.tryAddIMUReading(ctx, *reading.TimedIMUResponse); err != nil {
+			if errors.Is(err, cartofacade.ErrUnableToAcquireLock) {
+				config.Logger.Debugw("Skipping IMU sensor reading due to lock contention in cartofacade", "error", err)
+			} else {
+				config.Logger.Warnw("Skipping IMU sensor reading due to error from cartofacade", "error", err)
+			}
+		}
+	}
+
 	timeElapsedMs := int(time.Since(startTime).Milliseconds())
-	return int(math.Max(0, float64(1000/config.IMU.DataFrequencyHz()-timeElapsedMs)))
+	return int(math.Max(0, float64(1000/config.MovementSensor.DataFrequencyHz()-timeElapsedMs)))
 }
 
-// tryAddIMUReading tries to add a reading to the carto facade.
+// tryAddIMUReading tries to add an IMU reading to the carto facade.
 func (config *Config) tryAddIMUReading(ctx context.Context, reading s.TimedIMUReadingResponse) error {
-	err := config.CartoFacade.AddIMUReading(ctx, config.Timeout, config.IMU.Name(), reading)
+	err := config.CartoFacade.AddIMUReading(ctx, config.Timeout, config.MovementSensor.Name(), reading)
```

Comment:
Tested it, it's currently just the short name:

```bash
config.MovementSensor.Name():  replay-odom
```

What is preferred, and what are the consequences of using one vs. the other?

Reply:
Short name is preferred as it usually matches the name of the dependency that you're passing into the config. The full name is prepended with more information abut the namespace and family, which could be anything for a module.

I was just thrown off because the resource manager's API uses `Name` for the full name and `ShortName` of the shortname, so needed to make sure what this was.